arrows/vxl/convert_image.cxx

```diff
@@ -0,0 +1,499 @@
+  }
+  else if( scale_factor == 0.0 )
+  {
+    vil_convert_cast( input, output );
+  }
+  else
+  {
+    output = scale_image< opix_t >( input, scale_factor );
+  }
+  return output;
+}
+
+// ----------------------------------------------------------------------------
+convert_image
+::convert_image()
+  : d{ new priv{} }
+{
+  attach_logger( "arrows.vxl.convert_image" );
+}
+
+// ----------------------------------------------------------------------------
+convert_image
+::~convert_image()
+{
+}
+
+// ----------------------------------------------------------------------------
+vital::config_block_sptr
+convert_image
+::get_configuration() const
+{
+  // get base config from base class
+  vital::config_block_sptr config = algorithm::get_configuration();
+
+  config->set_value(
+    "format", d->format,
+    "Output type format: byte, sbyte, float, double, uint16, uint32, etc." );
+  config->set_value(
+    "single_channel", d->single_channel,
+    "Convert input (presumably multi-channel) to contain a single channel, "
+    "using either standard RGB to grayscale conversion weights, or "
+    "averaging." );
+  config->set_value(
+    "scale_factor", d->scale_factor,
+    "Optional input value scaling factor" );
+  config->set_value(
+    "random_greyscale", d->random_greyscale,
+    "Convert input image to a 3-channel greyscale image randomly with this "
+    "percentage between 0.0 and 1.0. This is used for machine learning "
+    "augmentation." );
+  config->set_value(
+    "percentile_norm", d->percentile_norm,
+    "If set, between [0, 0.5), perform percentile "
+    "normalization such that the output image's min and max "
+    "values correspond to the percentiles in the orignal "
+    "image at this value and one minus this value, respectively." );
+
+  return config;
+}
+
+// ----------------------------------------------------------------------------
+void
+convert_image
+::set_configuration( vital::config_block_sptr in_config )
+{
+  // Start with our generated vital::config_block to ensure that assumed values
+  // are present. An alternative would be to check for key presence before
+  // performing a get_value() call.
+  vital::config_block_sptr config = this->get_configuration();
+  config->merge_config( in_config );
+
+  // Settings for conversion
+  d->format = config->get_value< std::string >( "format" );
+  d->single_channel = config->get_value< bool >( "single_channel" );
+  d->scale_factor = config->get_value< double >( "scale_factor" );
+  d->random_greyscale = config->get_value< double >( "random_greyscale" );
+  d->percentile_norm = config->get_value< double >( "percentile_norm" );
+
+  // Adjustment in case user specified 1% instead of 0.01
+  if( d->percentile_norm >= 0.5 )
+  {
+    d->percentile_norm /= 100.;
+  }
+}
+
+// ----------------------------------------------------------------------------
+bool
+convert_image
+::check_configuration( vital::config_block_sptr config ) const
```

Comment:
```suggestion
::check_configuration( vital::config_block_sptr VTK_UNUSED config ) const
```

Reply:
It looks like this is typically placed before the type. And it's `VITAL_UNUSED`.

Reply:
Oops, sorry, muscle memory fail :flushed:. (`VTK_UNUSED` is IIRC `Type VTK_UNUSED(name)` which is why I also got the placement wrong; hopefully you caught that also on the other PR? Pretty sure I at least got the right macro name in the other one...)

Reply:
Both should be correctly placed.